tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -445,4 +445,177 @@ def test_consumer_groups(self):
 
         for ex in self._exceptions:
             if ex:
-                raise ex
\ No newline at end of file
+                raise ex
+
+    def _listdir(self, path, node, only_dirs=False):
```

Comment:
Avoid copying this function from redpanda.py -- move it into some shared location and re-use it.